velox/expression/tests/SimpleFunctionTest.cpp

```diff
@@ -1478,4 +1480,44 @@ TEST_F(SimpleFunctionTest, decimalsWithConstraints) {
   }
 }
 
+template <typename TExec>
+struct NoThrowFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  Status call(out_type<int64_t>& out, const arg_type<int64_t>& in) {
```

Comment:
maybe tests what happens if you have a runtime error? Should it always "throw" right away, or does "try" also swallow it?

Reply:
We don't have consistency here at the moment. TRY should suppress only USER errors. EvalCtx::applyToSelectedNoThrow throws non-USER errors and stores USER errors. However. EvalCtx::setError stores all errors. I think it makes sense for EvalCtx::setStatus to match EvalCtx::applyToSelectedNoThrow and store USER errors while throwing other errors. Let me know if you think otherwise.